lib/generateNix.js

```diff
@@ -24,28 +28,36 @@ const urlToName = require('./urlToName')
 //
 // to
 //
-// builtins.fetchGit {
-//   url = "https://1234user:1234pass@git.graphile.com/git/users/1234user/postgraphile-supporter.git";
-//   ref = "master";
-//   rev = "1234commit";
+// {
+//   name = "postgraphile-supporter.git";
+//   source = {
+//     type = "git";
+//     url = "https://1234user:1234pass@git.graphile.com/git/users/1234user/postgraphile-supporter.git";
+//     ref = "master";
+//     rev = "1234commit";
+//   };
 // }
 
 function fetchgit(fileName, url, rev, branch) {
-  return `    {
+  return `  {
     name = "${fileName}";
-    path =
-      let
-        repo = builtins.fetchGit {
-          url = "${url}";
-          ref = "${branch}";
-          rev = "${rev}";
-        };
-      in
-        runCommandNoCC "${fileName}" { buildInputs = [gnutar]; } ''
-          # Set u+w because tar-fs can't unpack archives with read-only dirs
-          # https://github.com/mafintosh/tar-fs/issues/79
-          tar cf $out --mode u+w -C \${repo} .
-        '';
+    source = {
+      type = "git";
+      url = "${url}";
+      ref = "${branch}";
+      rev = "${rev}";
+    };
+  }`
```

Comment:
there is an opportunity to return JSON data here and process the result with a JSON2nix converter function higher in the call stack